client/project_rolling_release.go

```diff
@@ -0,0 +1,271 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// RollingReleaseStage represents a stage in a rolling release
+type RollingReleaseStage struct {
+	TargetPercentage int  `json:"targetPercentage"`          // Required: 0-100
+	Duration         *int `json:"duration,omitempty"`        // Required for automatic advancement: 1-10000 minutes
+	RequireApproval  bool `json:"requireApproval,omitempty"` // Only in response for manual-approval type
+}
+
+// RollingRelease represents the rolling release configuration
+type RollingRelease struct {
+	Enabled         bool                  `json:"enabled"`         // Required
+	AdvancementType string                `json:"advancementType"` // Required when enabled=true: 'automatic' or 'manual-approval'
+	Stages          []RollingReleaseStage `json:"stages"`          // Required when enabled=true: 2-10 stages
+}
+
+// ErrorResponse represents the error response from the Vercel API
+type ErrorResponse struct {
+	Error struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+	} `json:"error"`
+}
+
+// Validate checks if the rolling release configuration is valid according to API requirements
```

Comment:
We can use validators on the terraform schema - by the time it's here it's effectively trusted.

Reply:
(or the API will return an error anyway).